tests/regression_tests/regression/test_springs.cpp

```diff
@@ -0,0 +1,58 @@
+#include <gtest/gtest.h>
+
+#include "test_utilities.hpp"
+
+#include "src/elements/springs/create_springs.hpp"
+#include "src/elements/springs/springs.hpp"
+#include "src/model/model.hpp"
+
+namespace openturbine::tests {
+
+inline auto SetUpSprings() {
+    auto model = Model();
+    // Add two nodes for the spring element
+    model.AddNode(
+        {0, 0, 0, 1, 0, 0, 0},  // First node at origin
+        {0, 0, 0, 1, 0, 0, 0}, {0, 0, 0, 0, 0, 0}, {0, 0, 0, 0, 0, 0}
+    );
+    model.AddNode(
+        {1, 0, 0, 1, 0, 0, 0},  // Second node at (1,0,0)
+        {0, 0, 0, 1, 0, 0, 0}, {0, 0, 0, 0, 0, 0}, {0, 0, 0, 0, 0, 0}
+    );
+
+    const auto springs_input = SpringsInput({SpringElement(
+        std::array{model.GetNode(0), model.GetNode(1)},
+        1000.,  // Spring stiffness
+        1.      // Undeformed length
+    )});
+
+    return CreateSprings(springs_input);
+}
+
+TEST(SpringsTest, NodeStateIndices) {
+    const auto springs = SetUpSprings();
+    EXPECT_EQ(springs.node_state_indices(0, 0), 0);
+    EXPECT_EQ(springs.node_state_indices(0, 1), 1);
```

Comment:
These checks access Device data on host and segfault for GPU builds.  Copy them down to host first.

Reply:
Could you confirm if the changes fixed the segfault?

Reply:
It works now!